meerkat_libs/db_adapters/postgresql.py

```diff
@@ -0,0 +1,246 @@
+import psycopg2
+import logging
+import json
+from psycopg2.extensions import ISOLATION_LEVEL_AUTOCOMMIT
+from psycopg2 import sql
+from psycopg2.extras import Json
+
+
+class PostgreSQLAdapter():
+
+    def __init__(self, db_name, structure, user='postgres', host='db'):
+        self.db_name = db_name
+        self.structure = structure
+        self.user = user
+        self.host = host
+        self.keys = self._extract_keys()
+
+    def drop(self):
+        try:
+            logging.info("Dropping tables")
+            conn = psycopg2.connect(
+                dbname=self.db_name,
+                user=self.user,
+                host=self.host
+            )
+
+            for table in self.structure.keys():
+                conn.cursor().execute("DROP TABLE {};".format(table))
+                conn.commit()
+
+            conn.close()
+
+        except psycopg2.ProgrammingError:
+            logging.info("No tables exists to drop.")
+
+    def setup(self):
+        self._create_db()
+        self._create_tables()
+
+    def list(self):
+        for table in self.structure.keys():
+            logging.info("Listing all records in {} table".format(table))
+            cur = self.conn().cursor()
+            cur.execute(sql.SQL("SELECT data from {}").format(
+                sql.Identifier(table)
+            ))
+            results = [x[0] for x in cur.fetchall()]
```

Comment:
Could you try to change the name of `x` and describe what it actually is? It's hard to guess what `x[0]` might be.